### PR TITLE
Terraform course added in courses/free-courses-en.md

### PR DESCRIPTION
**Pull request by Manish-Syal123**
## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo
Add resources
### Why is this valuable (or not)?
It helps engineers automate the process of provisioning their cloud and hybrid infrastructure. With it, we no longer need to go into the cloud provider console - AWS, Azure or GCP - and set everything up manually.
### How do we know it's really free?
The 2 courses are available free on youtube and other 4 are free Udemy courses.
## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -1141,6 +1142,16 @@
 * [Vapor University](https://vapor.university)
 
 
+### Terraform
+
+* [terraform + AWS](https://www.udemy.com/course/terraform-aws) - Rohit Abraham (Udemy)
+* [Terraform + GCP](https://www.udemy.com/course/terraform-gcp) - Rohit Abraham (Udemy)
+* [Terraform 101](https://www.udemy.com/course/terraform-101) - Jacob Jones (Udemy)
+* [Terraform on Azure - Basic Tutorial](https://www.udemy.com/course/terraform-on-azure-basic-tutorial) - Rahul Sawant (Udemy)
+* [Terraform tutorial for beginners Videos in English by Techworld with Murali](https://www.youtube.com/playlist?list=PLuJTeEt6HH9fUj7oRRLT0g1ttVl--Fv1T) - Murali (Youtube)
```
Comment (shrn01) [inline courses/free-courses-en.md:1151]:
```suggestion
* [Terraform tutorial for beginners Videos in English by Techworld with Murali](https://www.youtube.com/playlist?list=PLuJTeEt6HH9fUj7oRRLT0g1ttVl--Fv1T) - Murali
```

Youtube attribution is not required

**Comment by Manish-Syal123** [comment]
You mean that I Should remove (YouTube) from this.

**Review by shrn01 — APPROVED** [review]
LGTM

**Comment by Manish-Syal123** [comment]
Sorry But can you tell what this LGTM  means .

**Comment by eshellman** [comment]
"Looks Good To Me".
Thanks, and welcome to free-programming-books!

**Pull request merged**